src/onnxruntime.cc

```diff
@@ -2485,50 +2486,24 @@ ModelInstanceState::SetStringInputBuffer(
 
     size_t element_cnt = 0;
     if ((*responses)[idx] != nullptr) {
-      size_t remaining_bytes = expected_byte_size;
       char* data_content = input_buffer + buffer_copy_offset;
-      // Continue if the remaining bytes may still contain size info
-      while (remaining_bytes >= sizeof(uint32_t)) {
-        if (element_cnt >= expected_element_cnt) {
-          RESPOND_AND_SET_NULL_IF_ERROR(
-              &((*responses)[idx]),
-              TRITONSERVER_ErrorNew(
-                  TRITONSERVER_ERROR_INVALID_ARG,
-                  (std::string("unexpected number of string elements ") +
-                   std::to_string(element_cnt + 1) + " for inference input '" +
-                   input_name + "', expecting " +
-                   std::to_string(expected_element_cnt))
-                      .c_str()));
-          break;
-        }
-
-        const uint32_t len = *(reinterpret_cast<const uint32_t*>(data_content));
-        remaining_bytes -= sizeof(uint32_t);
+      TRITONSERVER_Error* err = ValidateStringBuffer(
+          data_content, expected_byte_size, expected_element_cnt,
+          input_name.c_str(), &str_list);
```

Comment:
If there is an error can we fail all the responses here rather than L2502.

Reply:
Two  `if (err != nullptr)`?
```
      TRITONSERVER_Error* err = ValidateStringBuffer(
          data_content, expected_byte_size, expected_element_cnt,
          input_name.c_str(), &str_list);
      if (err != nullptr) {
        RESPOND_AND_SET_NULL_IF_ERROR(&((*responses)[idx]), err);
      }

      // Set string values.
      for (const auto& [addr, len] : str_list) {
        // Make first byte of size info 0, so that if there is string data
        // in front of it, the data becomes valid C string.
        *const_cast<char*>(addr - sizeof(uint32_t)) = 0;
        string_ptrs->push_back(addr);
      }

      size_t element_cnt = str_list.size();
      if (err != nullptr) {
        FillStringData(string_ptrs, expected_element_cnt - element_cnt);
      }
```